serial.ts

```diff
@@ -121,34 +121,40 @@ class UsbEndpointUnderlyingSource implements UnderlyingByteSource {
    *
    * @param {ReadableByteStreamController} controller
    */
-  pull(controller: ReadableByteStreamController): void {
-    (async (): Promise<void> => {
```

Comment:
We still want to do the IIFE trick so that `pull()` resolves immediately so that a `read()` can be cancelled.

Reply:
I tried. don't know why, but with IIFE, chrome stopped responsing.

Reply:
I just tried this and couldn't reproduce this problem. This behavior usually means that you've introduced an infinite loop in your Javascript code which causes the page to become unresponsive. It should cause the browser to lock up.

Can you investigate this a little further?

Try adding a small timeout before issuing another `transferIn()` call so that if the device is responding with an endless stream of ZLPs it doesn't spin requesting more.